### PR TITLE
Fixing squid: S1168 Empty arrays and collections should be returned part 3

### PR DESCRIPTION
**Pull request by devFozgul**
This pull request is focused on resolving occurrences of Sonar rule 
 squid:S1168- “  Empty arrays and collections should be returned”. 
This TD will remove 60min TD.
 You can find more information about the issues here: 
 https://dev.eclipse.org/sonar/rules/show/squid:S1168
 Please let me know if you have any questions.
Fevzi Ozgul

### CODE REVIEW

**Comment by marcosnils** [comment]
Tests are not passing

**Comment by devFozgul** [comment]
Yes, exactly. I checked the logs where the tests failed. This PR required
quite a lot changes to be made in test methods. So I will do it in a more
convenient time. Problem is the assertNull methods for lists. Instead of
assertNull we might be use  Hamcrest empty list checking methods.

**Comment by marcosnils** [comment]
Perfect, let's leave it open until we can get tests to pass. Thx for the contribution!.

**Comment by devFozgul** [comment]
You are welcome. I have a couple of more issues to fix with this
project.However, they are not costly (TD wise) as the ones I have  already
done. You have also another "jedis something"  project  ( I do not remeber
the exact name now ) which I  will be working on in the coming days. I have
not checked it yet whether it has a considerable amount of TD to work on,
we shall see.

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 1]
```diff
@@ -168,7 +169,7 @@ private static Object process(final RedisInputStream is) {
   private static byte[] processBulkReply(final RedisInputStream is) {
     final int len = is.readIntCrLf();
     if (len == -1) {
-      return null;
+      return new byte[0];
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java]: [Redis Bulk String](https://redis.io/topics/protocol#resp-bulk-strings)
> The client library API should not return an empty string, but a nil object, when the server replies with a Null Bulk String.

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 2]
```diff
@@ -194,7 +195,7 @@ private static Long processInteger(final RedisInputStream is) {
   private static List<Object> processMultiBulkReply(final RedisInputStream is) {
     final int num = is.readIntCrLf();
     if (num == -1) {
-      return null;
+      return Collections.emptyList();
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java]: [Redis Arrays](https://redis.io/topics/protocol#resp-arrays)
> A client library API should return a null object and not an empty Array when Redis replies with a Null Array. This is necessary to distinguish between an empty list and a different condition.

**Pull request closed without merging**